File: diss/code_snippets/sprinkler.ml
let sprinkler_model =
  let* cloudy   = bernoulli .8 in
  let* rain     = bernoulli 
   (if cloudy then .8 else .1) in
  let* sprinkler = bernoulli 
   (if cloudy then .1 else .5) in
  let* wet_grass = bernoulli 
    (match rain,sprinkler with
         true,true -> .99
       | true,false -> .9
       | false,true -> .9
       | false,false -> 0.) in
  condition wet_grass
    (return rain)

File: code_snippets/webppl/sprinkler.js
var sprinkler_model = function (method) {
  return function () {
    var sprinkler = function () {
      var cloudy = bernoulli({ p: 0.5 });
      var rain = cloudy ? 
             bernoulli({ p: 0.8 }) 
           : bernoulli({ p: 0.2 });
      var sprinkler = cloudy ? 
             bernoulli({ p: 0.1 }) 
           : bernoulli({ p: 0.5 });
      var wet_grass = rain || sprinkler;
      condition(wet_grass);
      return rain;
    };
    return Infer(method, sprinkler);
  };
};
